sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -3982,6 +3982,14 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
       messageParameters = Map("provider" -> name))
   }
 
+  def externalDataSourceException(cause: Option[Throwable]): Throwable = {
```

Comment:
Let's don't overcomplicate the signature. If the caller pass `Throwable` always, then let's adjust the signature:
```suggestion
  def externalDataSourceException(cause: Throwable): Throwable = {
```

Reply:
fixed